src/Apps/ArtQuiz/ArtQuizContext/ArtQuizContext.tsx

```diff
@@ -0,0 +1,169 @@
+import {
+  createContext,
+  FC,
+  useCallback,
+  useContext,
+  useEffect,
+  useState,
+} from "react"
+import { extractNodes } from "Utils/extractNodes"
+
+// TODO: Kill all the anys!!
+export interface ArtQuizContextValues {
+  artworks: any[]
+  artworksTotalCount: number
+  currentArtwork: any | null
+  currentIndex: number
+  markItemComplete: (
+    artworkSlug: keyof ArtQuizContextValues["quizTemplate"]
+  ) => void
+  nextArtwork: any | null
+  previousArtwork: any | null
+  stepForward: () => void
+  stepBackward: () => void
+  quizTemplate: Record<string, boolean>
+}
+
+export const artQuizContext = createContext<ArtQuizContextValues>({
+  artworks: [],
+  artworksTotalCount: 0,
+  currentArtwork: null,
+  currentIndex: 0,
+  nextArtwork: null,
+  previousArtwork: null,
+  markItemComplete: (_: keyof ArtQuizContextValues["quizTemplate"]) => {
+    throw new Error("ArtQuizContext not initialized")
+  },
+  stepForward: () => {
+    throw new Error("ArtQuizContext not initialized")
+  },
+  stepBackward: () => {
+    throw new Error("ArtQuizContext not initialized")
+  },
+  quizTemplate: {},
+})
+
+export const ArtQuizContextProvider: FC = ({ children }) => {
+  // Quiz template should be the source of truth for the content and state of a
+  // particular user's quiz. A stringified version will be stored in a table
+  // with the userId as its key.
+  const [quizTemplate, setQuizTemplate] = useState<Record<string, boolean>>(
+    (undefined as unknown) as Record<string, boolean>
+  )
+
+  // This typing should clarify itself once we create our query and figure
+  // out which values we need to appropriately render the UI
+  const [artworks, _setArtworks] = useState<any[]>(extractNodes({ edges }))
```

Comment:
I wouldn't recommend storing actual artwork objects here. Fragments are always tied to specific pieces of UI. You shouldn't be passing them around. Should just store the index that points to one (even then probably an abstract cursor).

Reply:
Great point. Honestly, still kind of iffy on whether or not we actually need this context at all. May just shift all this into the ArtQuizApp component itself. I'll keep this in mind tho.